base_custom_filter/views/ir_filters_views.xml

```diff
@@ -9,11 +9,25 @@
         <field name="arch" type="xml">
             <field name="active" position="after">
                 <field name="type" />
+                <field
+                    name="search_field_id"
+                    domain="[('model', '=', model_id)]"
+                    attrs="{'required': [('type', '=', 'search')], 'invisible': [('type', '!=', 'search')]}"
+                />
+                <field
+                    name="filter_domain"
+                    attrs="{'invisible': [('type', '!=', 'search')]}"
+                />
+                <field
+                    name="group_ids"
+                    widget="many2many_tags"
+                    attrs="{'invisible': [('type', '=', 'favorite')]}"
+                />
                 <field
                     name="group_id"
                     domain="[('model_id', '=', model_id), ('type', '=', type)]"
                     context="{'default_model_id': model_id, 'default_type': type}"
-                    attrs="{'invisible': [('type', '=', 'favorite')]}"
+                    attrs="{'invisible': [('type', 'in', ['favorite', 'search'])]}"
                 />
             </field>
             <field name="domain" position="attributes">
```

Comment:
Can we hide domain field for the search type as well? The widget doesn't work in the way it should unless we tweak a bit so that it looks at the model of the selected field, and this attribute is rarely used anyway.